.github/workflows/main.yml

```diff
@@ -0,0 +1,112 @@
+name: Test Build and Call With Fake Register
+
+on: [push]
+
+jobs:
+  fakeRegister:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y openssl
+
+      - name: Install Python Dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install requests
+
+      - name: Docker Build
+        run: |
+          ls -la
+          docker-compose build
+
+      - name: Create Dummy Certificates
+        run: |
+          mkdir -p ./data/certs
+          openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:2048 -out ./data/certs/local-ca.key
+          openssl req -x509 -new -nodes -key ./data/certs/local-ca.key -sha256 -days 3650 -out ./data/certs/local-ca.pem -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          openssl req -x509 -new -nodes -key ./data/certs/local-ca.key -sha256 -days 3650 -out ./data/certs/factory_ca.pem -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:2048 -out ./data/certs/server.key
+          openssl req -new -key ./data/certs/server.key -out ./data/certs/server.csr -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          echo subjectAltName=DNS:www.example.com,DNS:example.org > ./data/certs/san.cnf
+          openssl x509 -req -in ./data/certs/server.csr -CA ./data/certs/local-ca.pem -CAkey ./data/certs/local-ca.key -CAcreateserial -out ./data/certs/tls-crt -days 365 -sha256 -extfile ./data/certs/san.cnf
+          rm -rf ./data/certs/server.key ./data/certs/server.csr ./data/certs/san.cnf
+          chmod +r ./data/certs/*   
+
+      - name: Run Docker Compose Up
+        run: |
+          docker-compose up --force-recreate --detach
```

Comment:
Does it help to simplify the below steps if we use e.g. `--wait --wait-timeout 30` here.

Reply:
I don't believe that's sufficient, as it doesn't guarantee that docker-compose is operational and accessible. Having this check in place allows us to quickly determine if there's an issue with the docker_run.sh script or with the mocks for the dummy certificates by attempting to run docker-compose.

Therefore, we could consider adding a timeout, but I would prefer to maintain the check as it currently stands.

Reply:
That's OK. I just wanted to make your life easier and the size of the Bash script smaller.